mmv1/products/certificatemanager/IssuanceConfigs.yaml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'IssuanceConfigs'
+base_url: 'projects/{{project}}/locations/{{location}}/certificateIssuanceConfigs'
+create_url: 'projects/{{project}}/locations/{{location}}/certificateIssuanceConfigs?certificateIssuanceConfigId={{name}}'
+self_link: 'projects/{{project}}/locations/{{location}}/certificateIssuanceConfigs/{{name}}'
+immutable: true
+description: |
+  Certificate represents a HTTP-reachable backend for a Certificate.
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+docs: !ruby/object:Provider::Terraform::Docs
+autogen_async: true
+import_format: ["projects/{{project}}/locations/{{location}}/certificateIssuanceConfigs/{{name}}"]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'certificate_manager_issuance_configs'
+    primary_resource_id: 'default'
+    vars:
+      ca_name: 'my-ca'
+      pool_name: 'my-ca-pool'
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    description: |
+      A user-defined name of the certificate issuance config.
+      CertificateIssuanceConfig names must be unique globally
+      and match pattern projects/*/locations/*/certificateIssuanceConfigs/*.
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      The Certificate Manager location. If not specified, "global" is used.
+    default_value: global
+    url_param_only: true
+properties:  
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      One or more paragraphs of text description of a CertificateIssuanceConfig.
+  - !ruby/object:Api::Type::Integer
+    name: 'rotationWindowPercentage'
+    description: |
+      It specifies the percentage of elapsed time of the certificate lifetime to wait before renewing the certificate. 
+      Must be a number between 1-99, inclusive. Defaults to 66. 
```

Comment:
The API definition looks different- the field is required and has no default. Can we mirror that?

![image](https://github.com/GoogleCloudPlatform/magic-modules/assets/2080520/9fcef8ad-c1dc-4db2-893f-798a38df57d8)

Reply:
Basically, I was following both the api documentation and gcloud certificate-manager documentation for creating CertificateIssuanceConfig (`gcloud certificate-manager issuance-configs create --help`).

There was a default value for `rotationWindowPercentage`, `lifetime` and the `keyAlgorithm`. That's why I did the same. 

So, generally. Are we supposed to be always consistent with only the API documentation? What is the best practice?

Reply:
Yeah, we should be consistent with the API rather than with other tools. Some accommodations like default values that make a lot of sense for gcloud make less sense for us. Terse resource definitions work great for interactive use, and explicit ones are more clear for config users.